drivers/flash/flash_simulator.c

```diff
@@ -380,6 +381,35 @@ flash_sim_get_parameters(const struct device *dev)
 	return &flash_sim_parameters;
 }
 
+#ifdef CONFIG_FLASH_EX_OP_ENABLED
+int flash_sim_ex_op(const struct device *dev, uint16_t code, const uintptr_t in, void *out)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(in);
+
+	int ret = 0;
+
+	switch (code) {
+	case NAND_FLASH_IS_BAD_BLOCK: {
+		/* Block is never bad on flash simulator */
+		*(int *)out = 0;
+		break;
+	}
+
+	case NAND_FLASH_MARK_BAD_BLOCK: {
+		/* Blocks cannot be marked bad on flash simulator */
+		break;
+	}
```

Comment:
I think we should also return -ENOTSUP here, we should describe return values for the NAND_FLASH_MARK_BAD_BLOCK and code calling the op should properly address the ENTOSUP.

Reply:
I am also not sure whether Flash Simulator, in the current form, is able to properly pretend NAND devices.
I think we should alter it to be able to use it for testing FTL on the CI, which also means that we may have to add ability to actually support bad blocks, as well as ecc or user bits.
I am not sure whether this is job for this pr, nevertheless we will not be able to reliably test ftl without it.